### DISCUSSION TITLE
New UI Design

### DISCUSSION THREAD

**Discussion by rafaelmardojai** [rafaelmardojai/blanket#289] [General]
This is a discussion to shape the final form of the new UI for Blanket begin implemented in the `new-ui` branch (see #288).

The start point is the last version of the mock up by  Sam Hewitt: https://gitlab.gnome.org/Teams/Design/other-app-mockups/-/blob/master/blanket/blanket.png.

**Comment by rafaelmardojai** [comment]
### Current Concerns

I really like the new design concept, IMO using a grid view that let you see almost all sounds without scrolling is a really nice improvement.

However I have some concerns about the volume controls usability. Optimally I think the user should be able to change the volumes from the grid. 

I have implemented the two obvious posible solutions in the `new-ui` branch, they are: showing the controls in the grid or in a sound context menu. The context menu also serves to remove a custom sound, something that the mockup doesn't cover.

(You can enable the controls in grid from the view-more menu next to the play/pause button)

![Captura desde 2022-12-30 12-50-09](https://user-images.githubusercontent.com/6210397/210098878-66a852cb-2ce6-4273-8b83-8427257d0133.png)

![Captura desde 2022-12-29 20-29-57](https://user-images.githubusercontent.com/6210397/210098809-796dca99-6f2e-4c70-a3dc-e1363938c17f.png)

Obviously the `GtkScale`s in  the view looks a bit off. So if we want to do this we'll need something more fancy if posible. Some nice idea would be to implement a circular volume slider around the icons when active, but that will be hard to implement or even imposible with python bindings.

This is an small idea I had about having the volume control in the icon itself, with similar behavior than Amberol's waveform control. But I can't figure out how the sound activation/deactivation would work.

![blanket-mockup](https://user-images.githubusercontent.com/6210397/210099469-9fb9d2e7-9708-486f-938b-3639ed726cd4.png)

So I would like to hear some ideas and suggestions from actual designers.

----

PD: When testing the branch if you have some playback issues just move the general volume control, I'm working on a playback refactor locally.

**Reply by bertob** [reply]
I like the circle-as-slider idea, and I think it could work if you do something like "tap=enable/disable" and "slide=increase/decrease volume".

I'd also move the bar with the playback buttons to the bottom and make it an actual toolbar.

**Reply by rafaelmardojai** [reply]
@bertob got this for now:

![Captura desde 2023-03-14 19-39-04](https://user-images.githubusercontent.com/6210397/225173639-f217ec52-459a-4867-8618-926e7feb4d08.png)

I think this is the better I can do without touching custom widget drawing in Python. What do you think?

**Reply by bertob** [reply]
Looks good! Is there a branch?

One thing I'd maybe tweak here is the thickness of the sliders, they seema  bit too big compared to the rest of the UI.

**Reply by rafaelmardojai** [reply]
> Looks good! Is there a branch?

Updated the `new-ui` branch with the changes.
 
> One thing I'd maybe tweak here is the thickness of the sliders, they seema bit too big compared to the rest of the UI.

They were `12px` I reduced them to `9px`, better?

![Captura desde 2023-03-15 19-41-44](https://user-images.githubusercontent.com/6210397/225480420-cb265ef8-59a9-459e-b808-877036250f8e.png)

**Reply by bertob** [reply]
Looks good, but I feel like the touch targets are too small on the individual sliders (both in the previous and the thinner version). Maybe dragging horizontally anywhere on the tile could adjust the volume?

**Reply by rafaelmardojai** [reply]
I guess I could add some event controllers to manipulate the volume from anywhere on the tile, but I wouldn't be able to test that.

**Reply by rafaelmardojai** [reply]
@bertob For now I have increased the volume scale target (invisible area around), hopefully that makes easier for touch.

**Reply by bertob** [reply]
I haven't tested it on touch, but the latest commit feels pretty good to me in that regard, yeah.

**Reply by rafaelmardojai** [reply]
@bertob Ok nice. Will be merging the branch then.
